### PR TITLE
Relocate TCP failure error code into TCP stats object

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

- Moves `retransmits` into `tcp_stats_t`
- Moves `failure_reason` into `tcp_stats_t`
- Resizes `failure_reason` to `u16` so it fits into the existing padding, thus not changing the overall size of `tcp_stats_t` and `conn_t`. Only three error codes are used, so no concerns about the reduced size.
- Removes separate perf/ring buffer for failed connections because that information is now contained on the connection object
- Reorders fields of `conn_t` so that `conn_stats` is aligned to a cacheline boundary.

### Motivation

- Simplify adding failure information to connections. This removes a race condition between failed and closed connection perf/ring buffers.
- Maintain size of `conn_t` so that perf buffer batching (coming in future PR) can remain a configuration option.

### Describe how to test/QA your changes

Existing automated tests passed.

### CODE REVIEW

File: pkg/network/tracer/connection/ebpf_tracer.go [hunk 1]
```diff
@@ -120,6 +121,7 @@ var EbpfTracerTelemetry = struct {
 	telemetry.NewStatCounterWrapper(connTracerModuleName, "pid_collisions", []string{}, "Counter measuring number of process collisions"),
 	telemetry.NewCounter(connTracerModuleName, "iteration_dups", []string{}, "Counter measuring the number of connections iterated more than once"),
 	telemetry.NewCounter(connTracerModuleName, "iteration_aborts", []string{}, "Counter measuring how many times ebpf iteration of connection map was aborted"),
+	telemetry.NewCounter("network_tracer__tcp_failure", "closed_conn_flushed_cleaned", []string{}, "Counter measuring the number of conn_close_flushed entries cleaned in userspace"),
```
Comment (hmahmood) [inline pkg/network/tracer/connection/ebpf_tracer.go:124]: "network_tracer__tcp_failure" -> `connTracerModuleName`

**Review by hmahmood — APPROVED** [review]

File: pkg/network/tracer/connection/ebpf_tracer.go (hunk 1 above)
Comment (brycekahle) [reply pkg/network/tracer/connection/ebpf_tracer.go:124]: I kept the metric name the same as before, but if you are OK with the change, I can do it.

File: pkg/network/ebpf/c/tracer.c [hunk 2]
```diff
@@ -287,14 +286,16 @@ int BPF_BYPASSABLE_KPROBE(kprobe__tcp_close, struct sock *sk) {
...
     if (bpf_map_update_with_telemetry(conn_close_flushed, &t, &timestamp, BPF_NOEXIST, -EEXIST) == 0) {
-        cleanup_conn(ctx, &t, sk);
-        increment_telemetry_count(tcp_close_connection_flush);
+        __u16 tcp_failure_reason = 0;
         int err = 0;
         bpf_probe_read_kernel_with_telemetry(&err, sizeof(err), (&sk->sk_err));
         if (err == TCP_CONN_FAILED_RESET || err == TCP_CONN_FAILED_TIMEOUT || err == TCP_CONN_FAILED_REFUSED) {
             increment_telemetry_count(tcp_close_target_failures);
-            flush_tcp_failure(ctx, &t, err);
+            tcp_failure_reason = err;
```
Comment (akarpz) [inline pkg/network/ebpf/c/tracer.c:294]: can we pass `err` directly to `cleanup_conn` to get the cast?

Comment (brycekahle) [reply pkg/network/ebpf/c/tracer.c:294]: This isn't for casting purposes, but to not set `err` unless it was one of the three desired values.

**Review by akarpz — APPROVED** [review]

File: pkg/network/tracer/connection/ebpf_tracer.go (hunk 1 above)
Comment (akarpz) [reply pkg/network/tracer/connection/ebpf_tracer.go:124]: I'm ambivalent about this change as there is no `failure` telemetry namespace anymore

**Review by derekwbrown — APPROVED** [review]
approved for single WKIT file

**Pull request merged**